addons/account/models/account_account.py

```diff
@@ -830,14 +830,14 @@ def name_search(self, name='', domain=None, operator='ilike', limit=100):
 
             move_type_accounts = {
                 'out': ['income'],
-                'in': ['expense', 'asset'],
+                'in': ['expense', 'asset_fixed'],
             }
             move_type_prefix = move_type.split('_')[0]
             # search all account types if the search term contains a number
             digit_in_search_term = any(c.isdigit() for c in name)
-            internal_group_domain = [('internal_group', 'in', move_type_accounts.get(move_type_prefix, []))] if not digit_in_search_term else []
+            type_domain = [('account_type', 'in', move_type_accounts.get(move_type_prefix, []))] if not digit_in_search_term else []
 
-            domain = Domain.AND([search_domain, internal_group_domain, domain])
+            domain = Domain.AND([search_domain, type_domain, domain])
```

Comment:
Are we sure that we want to be restrictive here? Isn't the order by enough?

Reply:
There's an escape hatch if you know the account code (or use digits)
but there shouldn't be a need to use other accounts, but that's why you could search more 
(instead of a domain on the field)